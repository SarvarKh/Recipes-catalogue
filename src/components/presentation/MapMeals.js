import { Link } from 'react-router-dom';

const MapMeals = ({ meals, clickOnDetailMeal }) => (
  meals.map((meal) => (
    <Link
      to={{
        pathname: `/detail/${meal.idMeal}`,
        state: meal,
      }}
      key={meal.idMeal}
      className="meal"
      onClick={() => clickOnDetailMeal(meal.idMeal)}
    >
      <div className="img-container">
        <img src={meal.strMealThumb} alt="Meal" />
      </div>
      <h3>{meal.strMeal}</h3>
    </Link>
  ))
);

export default MapMeals;
