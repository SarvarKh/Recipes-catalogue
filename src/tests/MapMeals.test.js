import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import MapMeals from '../components/presentation/MapMeals';

const meals = [
    {
        idMeal: '88',
        strMeal: 'Beef',
        strMealThumb: 'beef',
    },
    {
        idMeal: '92',
        strMeal: 'Lamb',
        strMealThumb: 'lamb',
    },
];

const clickOnDetailMeal = () => (
    console.log('Test MapMeals and logging meal name')
)

it('renders correctly', () => {
  const tree = renderer
    .create(<BrowserRouter><MapMeals meals={meals} clickOnDetailMeal={clickOnDetailMeal} /></BrowserRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});