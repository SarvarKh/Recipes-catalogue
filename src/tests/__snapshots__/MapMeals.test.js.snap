// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <a
    className="meal"
    href="./components/DetailMeal#88"
    onClick={[Function]}
  >
    <div
      className="img-container"
    >
      <img
        alt="Meal"
        src="beef"
      />
    </div>
    <h3>
      Beef
    </h3>
  </a>,
  <a
    className="meal"
    href="./components/DetailMeal#92"
    onClick={[Function]}
  >
    <div
      className="img-container"
    >
      <img
        alt="Meal"
        src="lamb"
      />
    </div>
    <h3>
      Lamb
    </h3>
  </a>,
]
`;
